src/components/Calculator.js

```diff
@@ -1,15 +1,40 @@
 /* eslint-disable react/prefer-stateless-function */
 import React from 'react';
-
-const buttons = ['AC', '+/-', '%', '÷', 7, 8, 9, 'x', 4, 5, 6, '-', 1, 2, 3, '+', 0, '.', '='];
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      operation: null,
+      next: null,
+      total: null,
+    };
+    this.btnClicked = this.btnClicked.bind(this);
+  }
+
+  btnClicked = (event) => {
+    const buttonName = event.target.innerText;
+    const result = calculate(this.state, buttonName);
+    this.setState((previousState) => {
+      const { operation, next, total } = previousState;
+      return {
+        operation: ((result.operation === undefined) ? operation : result.operation),
+        next: ((result.next === undefined) ? next : result.next),
+        total: ((result.total === undefined) ? total : result.total),
+      };
+    });
+  };
+
   render() {
+    const buttons = ['AC', '+/-', '%', '÷', 7, 8, 9, 'x', 4, 5, 6, '-', 1, 2, 3, '+', 0, '.', '='];
+    const { operation, next, total } = this.state;
+    const display = (total || '') + (operation || '') + (next || '');
     return (
       <div className="calculator">
-        <p className="result">0</p>
+        <p className="result">{display}</p>
         <div className="buttons">
-          {buttons.map((btn) => <button key={btn} type="button">{btn}</button>)}
+          {buttons.map((btn) => <button key={btn} type="button" onClick={this.btnClicked}>{btn}</button>)}
```

Comment:
- [ ] You are encouraged to use props while implementing the events. Currently, you have all the logic in one file but I would urge you to separate it in such a way that you have a child component. For example, as you have mapped through all the buttons above you could now call a<Button> component inside that map that will have the necessary props like key, type and onclick then pass the props to the button component in another file. This will make your work cleaner and neater.